cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/PredefinedAbstainDRep.hs

```diff
@@ -0,0 +1,357 @@
+import qualified Testnet.Process.Cli as P
+import qualified Testnet.Process.Run as H
+import qualified Testnet.Property.Utils as H
+import           Testnet.Runtime
+
+import           Hedgehog
+import qualified Hedgehog.Extras as H
+import qualified Hedgehog.Extras.Stock.IO.Network.Sprocket as IO
+
+-- | This test creates a default testnet with three DReps and one stake holder delegated to each.
+-- We then do a proposal for an arbitrary parameter change (in this case to the
+-- @desiredNumberOfPools@ parameter) to check that it fails, when the first DRep votes "yes" and the
+-- last two vote "no". Later we chack that if we change the stake holders under the DReps that vote
+-- "no" to delegate to the automate "always abstain" DRep, the same kind of proposal passes.
+-- 
+-- This test is meant to ensure that delegating to "always abstain" has the desired effect of
+-- counting as abstaining for the stake delegated.
+--
+-- Execute me with:
+-- @DISABLE_RETRIES=1 cabal test cardano-testnet-test --test-options '-p "/Predefined Abstain DRep/"'@
+hprop_check_predefined_abstain_drep :: Property
+hprop_check_predefined_abstain_drep = H.integrationWorkspace "test-activity" $ \tempAbsBasePath' -> do
+  -- Start a local test net
+  conf@Conf { tempAbsPath } <- mkConf tempAbsBasePath'
+  let tempAbsPath' = unTmpAbsPath tempAbsPath
+      tempBaseAbsPath = makeTmpBaseAbsPath tempAbsPath
+
+  work <- H.createDirectoryIfMissing $ tempAbsPath' </> "work"
+
+  -- Create default testnet with 3 DReps and 3 stake holders delegated, one to each DRep.
+  let ceo = ConwayEraOnwardsConway
+      sbe = conwayEraOnwardsToShelleyBasedEra ceo
+      era = toCardanoEra sbe
+      cEra = AnyCardanoEra era
+      fastTestnetOptions = cardanoDefaultTestnetOptions
+        { cardanoEpochLength = 100
+        , cardanoNodeEra = cEra
+        , cardanoNumDReps = 3
+        }
+
+  testnetRuntime@TestnetRuntime
+    { testnetMagic
+    , poolNodes
+    , wallets=wallet0:wallet1:wallet2:_
+    , configurationFile
+    }
+    <- cardanoTestnetDefault fastTestnetOptions conf
+
+  poolNode1 <- H.headM poolNodes
+  poolSprocket1 <- H.noteShow $ nodeSprocket $ poolRuntime poolNode1
+  execConfig <- H.mkExecConfig tempBaseAbsPath poolSprocket1 testnetMagic
+
+  let socketName' = IO.sprocketName poolSprocket1
+      socketBase = IO.sprocketBase poolSprocket1 -- /tmp
+      socketPath = socketBase </> socketName'
+
+  epochStateView <- getEpochStateView (File configurationFile) (File socketPath)
+
+  startLedgerNewEpochStateLogging testnetRuntime tempAbsPath'
+
+  H.note_ $ "Sprocket: " <> show poolSprocket1
+  H.note_ $ "Abs path: " <> tempAbsBasePath'
+  H.note_ $ "Socketpath: " <> socketPath
+  H.note_ $ "Foldblocks config file: " <> configurationFile
+
+  gov <- H.createDirectoryIfMissing $ work </> "governance"
+
+  initialDesiredNumberOfPools <- getDesiredPoolNumberValue execConfig
+
+  let newNumberOfDesiredPools = fromIntegral (initialDesiredNumberOfPools + 1)
```

Comment:
`fromIntegral` seems redundant here

Reply:
Huh, you are right

Reply:
Fixed in: https://github.com/IntersectMBO/cardano-node/pull/5815/commits/0ca06c8d906cdb33bbc7cb3c49022ca9dba85dd0